src/app/clusters/diagnostic-logs-server/DiagnosticLogsProviderDelegate.h

```diff
@@ -59,9 +59,11 @@ class DiagnosticLogsProviderDelegate
      * returned from StartLogCollection.
      *
      * @param[in] sessionHandle  The unique handle for this log session returned from a call to StartLogCollection.
+     * @param[in] error          A CHIP_ERROR value indicating the reason for ending the log collection.
+     *                           It is permissible to pass CHIP_NO_ERROR to indicate normal termination.
      *
      */
-    virtual CHIP_ERROR EndLogCollection(LogSessionHandle sessionHandle) = 0;
+    virtual CHIP_ERROR EndLogCollection(LogSessionHandle sessionHandle, CHIP_ERROR error = CHIP_NO_ERROR) = 0;
```

Comment:
This is still a breaking change.  Like I asked: do we want to cause a break here, or do we want to do this in a backwards-compatible manner?  That question was not answered.

Reply:
@bzbarsky-apple I have overloaded the method in a backward-compatible manner. Please review.